server/http/controller.go

```diff
@@ -569,7 +569,7 @@ func (c *controller) executeTest(ctx context.Context, test model.Test, runInfo o
 	testID := id.ID(resp.Body.(openapi.UpsertDefinitionResponse).Id)
 	// test ready, execute it
 	resp, err = c.RunTest(ctx, testID.String(), runInfo)
-	if err != nil {
+	if resp.Code != http.StatusOK {
```

Comment:
Being paranoid: can we handle both status and error here? Like this:

```suggestion
	if resp.Code != http.StatusOK || err != nil {
```